src/System.Private.Uri/src/System/UriBuilder.cs

```diff
@@ -37,12 +33,12 @@ public UriBuilder(string uri)
 
             if (tryUri.IsAbsoluteUri)
             {
-                Init(tryUri);
+                SetFieldsFromUri(tryUri);
             }
             else
             {
                 uri = Uri.UriSchemeHttp + Uri.SchemeDelimiter + uri;
-                Init(new Uri(uri));
```

Comment:
Is this the kind of refactoring that @davidsh said should be avoided in this library?